service/worker/addsearchattributes/fx.go

```diff
@@ -69,16 +73,29 @@ func NewResult(params initParams) fxResult {
 	}
 }
 
-func (wc *addSearchAttributes) Register(worker sdkworker.Worker) {
+func (wc *addSearchAttributes) RegisterWorkflow(worker sdkworker.Worker) {
 	worker.RegisterWorkflowWithOptions(AddSearchAttributesWorkflow, workflow.RegisterOptions{Name: WorkflowName})
-	worker.RegisterActivity(wc.activities())
 }
 
-func (wc *addSearchAttributes) DedicatedWorkerOptions() *workercommon.DedicatedWorkerOptions {
+func (wc *addSearchAttributes) DedicatedWorkflowWorkerOptions() *workercommon.DedicatedWorkerOptions {
 	// use default worker
 	return nil
 }
 
+func (wc *addSearchAttributes) RegisterActivities(worker sdkworker.Worker) {
+	worker.RegisterActivity(wc.activities())
+}
+
+func (wc *addSearchAttributes) DedicatedActivityWorkerOptions() *workercommon.DedicatedWorkerOptions {
+	return &workercommon.DedicatedWorkerOptions{
+		TaskQueue: primitives.AddSearchAttributesActivityTQ,
+		Options: sdkworker.Options{
+			BackgroundActivityContext: headers.SetCallerType(context.Background(), headers.CallerTypeAPI),
+			DisableWorkflowWorker:     true,
```

Comment:
can't the worker manager set this part automatically if this returns non-nil? why would a "dedicated activity worker" ever run a workflow worker?